srtcore/api.cpp

```diff
@@ -1908,11 +1908,29 @@ int srt::CUDTUnited::close(const SRTSOCKET u)
         return 0;
     }
 #endif
-    CUDTSocket* s = locateSocket(u);
-    if (!s)
-        throw CUDTException(MJ_NOTSUP, MN_SIDINVAL, 0);
+#if ENABLE_HEAVY_LOGGING
+    // Wrapping the log into a destructor so that it
+    // is printed AFTER the destructor of SocketKeeper.
+    struct ForceDestructor
+    {
+        CUDTSocket* ps;
+        ForceDestructor(): ps(NULL){}
+        ~ForceDestructor()
+        {
+            if (ps) // Could be not acquired by SocketKeeper, occasionally
+            {
+                HLOGC(smlog.Debug, log << "CUDTUnited::close/end: @" << ps->m_SocketID << " busy=" << ps->isStillBusy());
+            }
+        }
+    } fod;
+#endif
+
+    SocketKeeper k(*this, u, ERH_THROW);
+    IF_HEAVY_LOGGING(fod.ps = k.socket);
+    HLOGC(smlog.Debug, log << "CUDTUnited::close/begin: @" << u << " busy=" << k.socket->isStillBusy());
+    int ret = close(k.socket);
 
-    return close(s);
+    return ret;
```

Comment:
Just `return close(k.socket);`.

Reply:
Likely I wanted to print this `ret` in some log, but nevermind.